ase/scripts/generate_ase_environment.py

```diff
@@ -509,6 +509,26 @@ def config_qsys_sources(filelist, vlog_srcs, vhdl_srcs):
     return qsys_sim_files
 
 
+#
+# Sort qsys generated files since the order they are passed to the
+# simulator may matter. In synth mode, qsys-generate doesn't emit
+# any configuration files that define compilation order, so we are
+# forced to use a heuristic.
+#
+def sort_key_qsys_files(fn):
+    # Sort by prepending an ordering token
+    root, ext = os.path.splitext(fn)
+    if (root.endswith('package') or root.endswith('pkg')):
+        return '0' + fn
+    if (root.endswith('library') or root.endswith('lib')):
+        return '1' + fn
+    if (root.endswith('functions')):
+        return '2' + fn
+    if (root.endswith('components')):
+        return '3' + fn
+    return '4' + fn
```

Comment:
Do qsys generated files follow the naming convention of pkg, lib, functions and components?

Reply:
The key files do. It's the best I could come up with.